### PR TITLE
📚 Adding CITATION.cff

### LINKED ISSUE #264: 📑 CITATION files
Just learned that we can place `CITATION.cff` at the root of the repository.

### PR DESCRIPTION
**Pull request by jonsterling**
resolve #264 

So I wasn't exactly sure what to put for authors, but I felt that there are two competing concerns that we can address simultaneously. I want to credit the RedPRL Development Team as a whole, which includes not only us who are pushing code but also non-coding contributors; but I feel that individuals who have made a significant contribution to the code as it exists today (and its design) should be credited in the citation. These individuals make up what I conceptualize as the 'core team' (myself, favonia, carlo, reed) and I have credited these by name.

### CODE REVIEW

File: CITATION.cff [hunk 1]
```diff
@@ -0,0 +1,22 @@
+cff-version: 1.1.0
+message: If you use this software, please cite it as below.
+authors:
+  - family-names: Sterling
+    given-names: Jonathan
+    orcid: https://orcid.org/0000-0002-0585-5564
+    website: https://www.jonmsterling.com/
+  - name: Favonia
```
Comment (cangiuli) [inline CITATION.cff:8]: The [CFF converter](https://github.com/citation-file-format/cff-converter-python) drops this. It seems that the only options are to use `given-names` or `alias`, or to adopt Favonia's suggested [LaTeX format](https://favonia.org/). @favonia?

**Review by cangiuli — CHANGES_REQUESTED** [review]
Relatedly, we should add an AFOSR ack to the README.

**Comment by RobertHarper** [comment]
Indeed.

Copyright (c) Robert Harper.  All Rights Reserved.

File: CITATION.cff (hunk 1 above)
Comment (favonia) [reply CITATION.cff:8]: For consistency I prefer the LaTeX format `Kuen-Bang {Hou (Favonia)}`. (But it seems I can see "Favonia" on the GitHub web page?)

**Review by favonia — APPROVED** [review]

**Review by favonia — APPROVED** [review]
I have checked my part. (My approval does not imply others are happy.)

**Review by cangiuli — APPROVED** [review]

**Comment by jonsterling** [comment]
since the bibtex doesn’t support the entities, is there a “note” field where we can credit the whole RedPRL team?

File: CITATION.cff [hunk 2]
```diff
@@ -0,0 +1,25 @@
+cff-version: 1.1.0
+message: If you use this software, please cite it as below.
+authors:
+  - family-names: Sterling
```
Comment (jonsterling) [inline CITATION.cff:4]: if we are doing quotes for the yaml, let’s be uniform. i don’t have any preference which way or the other.

Comment (favonia) [reply CITATION.cff:4]: I recommend quoting all values because who knows what would happen. To be pedantic, you can also quote all the keys.

**Comment by jonsterling** [comment]
I hvae been experimenting with the cff-to-bibtex conversion tool and I think that it is so poor that we would _not_ want anyone to use the citation it generates. Therefore, I believe we should not include a `CITATION.cff` file at this time, and should instead (optionally) provide a proper bibtex citation somewhere in the repo.

**Pull request closed without merging**